Implement `add.Scalar` as dispatch-less composite kernel.

ysiraichi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #77486
* __->__ #77485
* #77484

This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

ysiraichi committed:
Implement `add.Scalar` as dispatch-less composite kernel.

This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Update on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Fix lint errors. on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Update on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Update on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Fix lint. on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Fix lint. on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Rebased. on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Fix merge. on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

bdhirsh reviewed (approved):
LGTM!

@ysiraichi since all the extra codegen we're added is primarily there for the perf benefit, can you run a micro-benchmark for `add.Scalar`?

ysiraichi committed:
Revert 'add_' dispatch table. on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Rebased. on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Update on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Fix lint issues. on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Add 'DispatchlessDEV.cpp' to BAZEL builds. on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Rebased. on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Update on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi committed:
Fix lint. on "Implement `add.Scalar` as dispatch-less composite kernel."


This PR makes `add.Scalar` kernel into a dispatch-less composite kernel.
Here are the main changes:

- `BinaryOps.cpp`
    - Removed `add` function definitions (only for `add.Scalar`)
- `composite/add.h`
    - Add the templated functions
- `native_functions.yaml`
    - Add the `composite` key to `add.Scalar`

[ghstack-poisoned]

ysiraichi commented:
@bdhirsh Sure! The results are [here](https://github.com/pytorch/pytorch/pull/77484#issuecomment-1152204914)
